src/caldavclient.cpp

```diff
@@ -436,107 +453,62 @@ void CalDavClient::start()
-        syncFinished(Buteo::SyncResults::DATABASE_FAILURE, "unable to query mapped notebooks");
-        return;
-    }
-    Q_FOREACH (const QString &knownNotebookUid, knownNotebookUids) {
-        LOG_TRACE("Have known notebook:" << knownNotebookUid);
-    }
-
     mKCal::Notebook::List validNotebooks;
     for (int i = 0; i < notebooks.size(); ++i) {
         mKCal::Notebook::Ptr notebook = notebooks[i];
-        if (notebook->account().startsWith(QStringLiteral("%1-").arg(QString::number(mAccountId)))
-                || notebook->account() == QString::number(mAccountId)) {
+        if (notebook->account() == QString::number(mAccountId)) {
             // This notebook is for this account.
             LOG_TRACE("Have notebook:" << notebook->uid() << "for account:" << mAccountId << notebook->account());
-            // Note that the first check exists for historical reasons,
-            // as we used to store calendar-path information in the account id field.
-            if (knownNotebookUids.contains(notebook->uid())) {
-                // this is a valid, mapped notebook.
-                validNotebooks.append(notebook);
-            } else {
-                // purge this notebook, as we no longer track it in our OOB db.
-                LOG_DEBUG("found old notebook:" << notebook->uid() << "for account" << mAccountId << "; purging");
-                if (!deleteNotebook(mAccountId, mCalendar, mStorage, notebook)) {
-                    LOG_DEBUG("failed to purge unmapped notebook:" << notebook->uid());
-                    syncFinished(Buteo::SyncResults::DATABASE_FAILURE, "failed to purge unmapped notebook");
-                    return;
-                }
-            }
+            validNotebooks.append(notebook);
         }
     }
 
     // for each calendar path we need to sync:
-    //  - if it is mapped to a known notebook, we need to perform either clean sync or quick sync
+    //  - if it is mapped to a known notebook, we need to perform quick sync
     //  - if no known notebook exists for it, we need to create one and perform clean sync
     Q_FOREACH (const Settings::CalendarInfo &calendarInfo, allCalendarInfo) {
-        QString notebookUidToDelete;
         mKCal::Notebook::Ptr existingNotebook;
         Q_FOREACH (mKCal::Notebook::Ptr notebook, validNotebooks) {
-            if (mDatabase->remoteCalendarPath(notebook->uid(), &ok) == calendarInfo.serverPath) {
-                LOG_DEBUG("found notebook:" << notebook->uid() << "for remote calendar:" << calendarInfo.serverPath);
+            // we abuse the pluginName() field in mKCal::Notebook to store not just the plugin name
+            // but also the remote calendar path, because Notebook API is deficient and doesn't have
+            // a dedicated field for the remote calendar path url.
+            if (notebook->syncProfile().endsWith(QStringLiteral(":%1").arg(calendarInfo.remotePath))) {
```

Comment:
Could expect plugin name being same for all notebooks handled by one plugin. Then again don't know what sync profile actually is, and neither is it specified, first thing coming to mind from the name is something telling when to sync. But a string is strange data type for such information, so maybe we could just abuse that.